### PR TITLE
docs(redis source): Use generated docs for configuration

### LINKED ISSUE #16075: Convert the `redis` source to use auto-generated docs

### PR DESCRIPTION
**Pull request by StephenWakely**
Closes #16075 
Epic #14998

Signed-off-by: Stephen Wakely <fungus.humungus@gmail.com>

### CODE REVIEW

File: src/sources/redis/mod.rs [hunk 1]
```diff
@@ -108,16 +108,19 @@ pub struct RedisSourceConfig {
...
     /// The Redis key to read messages from.
+    #[configurable(metadata(docs::examples = "vector"))]
     key: String,
 
     /// Sets the name of the log field to use to add the key to each event.
     ///
     /// The value will be the Redis key that the event was read from.
     ///
     /// By default, this is not set and the field will not be automatically added.
+    #[configurable(metadata(docs::examples = "redis_key"))]
```
Comment (neuronull) [inline src/sources/redis/mod.rs:123]: 🤔  Is it just me, or is the combination of the doc comments for these two members, confusing?

Is the value taken from "key" here, the value referred to by the comment in "redis_key" ?

Comment (spencergilbert) [reply src/sources/redis/mod.rs:123]: No, `redis_key` determines what event path data is added - `key` is the Redis key/value setup.

It's a difficult naming situation.

File: src/sources/redis/mod.rs [hunk 2]
```diff
@@ -108,16 +108,19 @@ pub struct RedisSourceConfig {
     /// The Redis URL to connect to.
     ///
     /// The URL must take the form of `protocol://server:port/db` where the `protocol` can either be `redis` or `rediss` for connections secured via TLS.
+    #[configurable(metadata(docs::examples = "redis://127.0.0.1:6379/0"))]
```
[the remark this answers is not in the document]
Comment (StephenWakely) [reply src/sources/redis/mod.rs:111]: Yes. Redis databases are a number 0-16 (I think). Database 0 is the default.

[the remark this answers is not in the document]
Comment (spencergilbert) [reply src/sources/redis/mod.rs:111]: Cool, been a while since I had to work with redis

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**